lib/api/unittest/CDataSummarizationTest.cc

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+#include <core/CBase64Filter.h>
+
+#include <maths/CBoostedTreeLoss.h>
+
+#include <api/CDataFrameAnalyzer.h>
+#include <api/CDataSummarization.h>
+
+#include <test/CDataFrameAnalysisSpecificationFactory.h>
+#include <test/CDataFrameAnalyzerTrainingFactory.h>
+#include <test/CRandomNumbers.h>
+
+#include <rapidjson/document.h>
+#include <rapidjson/schema.h>
+
+#include <boost/iostreams/copy.hpp>
+#include <boost/iostreams/filter/gzip.hpp>
+#include <boost/iostreams/filtering_stream.hpp>
+#include <boost/test/unit_test.hpp>
+
+#include <fstream>
+#include <string>
+
+BOOST_AUTO_TEST_SUITE(CDataSummarizationTest)
+
+using namespace ml;
+
+namespace {
+using TDoubleVec = std::vector<double>;
+using TStrVec = std::vector<std::string>;
+using TStrVecVec = std::vector<TStrVec>;
+using TFilteredInput = boost::iostreams::filtering_stream<boost::iostreams::input>;
+using TLossFunctionType = maths::boosted_tree::ELossType;
+
+std::stringstream decompressStream(std::stringstream&& compressedStream) {
+    std::stringstream decompressedStream;
+    {
+        TFilteredInput inFilter;
+        inFilter.push(boost::iostreams::gzip_decompressor());
+        inFilter.push(core::CBase64Decoder());
+        inFilter.push(compressedStream);
+        boost::iostreams::copy(inFilter, decompressedStream);
+    }
+    return decompressedStream;
+}
+
+void testIntegration(TLossFunctionType lossType) {
```

Comment:
This seems more appropriate maybe:
```suggestion
void testSchema(TLossFunctionType lossType) {
```

Reply:
Renamed